workspaces/arborist/lib/add-rm-pkg-deps.js

```diff
@@ -4,8 +4,67 @@ const log = require('proc-log')
 const localeCompare = require('@isaacs/string-locale-compare')('en')
 
 const add = ({ pkg, add, saveBundle, saveType }) => {
-  for (const spec of add) {
-    addSingle({ pkg, spec, saveBundle, saveType })
+  for (const { name, rawSpec } of add) {
+    // if the user does not give us a type, we infer which type(s)
+    // to keep based on the same order of priority we do when
+    // building the tree as defined in the _loadDeps method of
+    // the node class.
+    if (!saveType) {
+      saveType = inferSaveType(pkg, name)
+    }
```

Comment:
instead of this, maybe `saveType = inferSaveType(pkg, name)` in the signature? unless you expect a user to give you a non-undefined falsy value that you want to preserve (like an empty string)

Reply:
that's outside the scope of this PR, it was just inlining existing code